datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -126,111 +127,62 @@ macro_rules! get_statistic {
     }};
 }
 
-#[derive(Debug, Clone, Copy)]
-enum Statistic {
-    Min,
-    Max,
-}
-
-/// Extracts statistics for a single leaf column from [`RowGroupMetaData`] as an
-/// arrow [`ArrayRef`]
+/// Lookups up the parquet column by name
 ///
-/// For example, given a parquet file with 3 Row Groups, when asked for
-/// statistics for column "A" it will return a single array with 3 elements.
-pub(crate) struct RowGroupStatisticsConverter<'a> {
-    field: &'a Field,
+/// Returns the parquet column index and the corresponding arrow field
+pub fn parquet_column<'a>(
+    parquet_schema: &SchemaDescriptor,
+    arrow_schema: &'a Schema,
+    name: &str,
+) -> Option<(usize, &'a FieldRef)> {
+    let (root_idx, field) = arrow_schema.fields.find(name)?;
+    if field.data_type().is_nested() {
+        // Nested fields are not supported and require non-trivial logic
+        // to correctly walk the parquet schema accounting for the
+        // logical type rules - <https://github.com/apache/parquet-format/blob/master/LogicalTypes.md>
+        //
+        // For example a ListArray could correspond to anything from 1 to 3 levels
+        // in the parquet schema
+        return None;
+    }
+
+    // This could be made more efficient (#TBD)
+    let parquet_idx = (0..parquet_schema.columns().len())
+        .find(|x| parquet_schema.get_column_root_idx(*x) == root_idx)?;
+    Some((parquet_idx, field))
 }
 
-impl<'a> RowGroupStatisticsConverter<'a> {
-    /// Create a new RowGoupStatisticsConverter for extracting the named column
-    /// in the schema.
-    pub fn try_new(schema: &'a arrow_schema::Schema, column_name: &str) -> Result<Self> {
-        let field = schema.field_with_name(column_name)?;
-        Ok(Self { field })
-    }
-
-    /// Returns the min value for the column into an array ref.
-    pub fn min<'b>(
-        &self,
-        row_group_meta_data: impl IntoIterator<Item = &'b RowGroupMetaData>,
-    ) -> Result<ArrayRef> {
-        self.min_max_impl(Statistic::Min, row_group_meta_data)
-    }
-
-    /// Returns the max value for the column into an array ref.
-    pub fn max<'b>(
-        &self,
-        row_group_meta_data: impl IntoIterator<Item = &'b RowGroupMetaData>,
-    ) -> Result<ArrayRef> {
-        self.min_max_impl(Statistic::Max, row_group_meta_data)
-    }
-
-    /// Extracts all min/max values for the column into an array ref.
-    fn min_max_impl<'b>(
-        &self,
-        mm: Statistic,
-        row_group_meta_data: impl IntoIterator<Item = &'b RowGroupMetaData>,
-    ) -> Result<ArrayRef> {
-        let mut row_group_meta_data = row_group_meta_data.into_iter().peekable();
-
-        // if it is empty, return empty array
-        if row_group_meta_data.peek().is_none() {
-            return Ok(new_empty_array(self.field.data_type()));
-        }
+/// Extracts the min statistics from an iterator of [`ParquetStatistics`] to an [`ArrayRef`]
+pub fn min_statistics<'a, I: Iterator<Item = Option<&'a ParquetStatistics>>>(
```

Comment:
Originally I had this as ColumnChunkMetadata, however, we don't actually need anything beyond the ParquetStatistics, so it seemed peculiar to require the full `ColumnChunkMetadata`. Additionally one method to support the column index using the same array logic would be to coerce both kinds of statistics to the same representation